src/parse.rs

```diff
@@ -22,9 +23,64 @@ pub trait FromHex: Sized {
     where
         I: Iterator<Item = Result<u8, HexToBytesError>> + ExactSizeIterator + DoubleEndedIterator;
 
+    /// Parses provided string as hex explicitly requiring there to not be a `0x` prefix.
+    ///
+    /// This is not recommended for user-supplied inputs because of possible confusion with decimals.
+    /// It should be only used for existing protocols which always encode values as hex without 0x prefix.
+    #[rustfmt::skip]
+    fn from_hex<
+        #[cfg(feature = "alloc")] S: AsRef<str> + Into<String>,
+        #[cfg(not(feature = "alloc"))] S: AsRef<str>,
+    >(s: S) -> Result<Self, Self::Error> {
+        Self::from_no_prefix_hex(s)
+    }
+
     /// Produces an object from a hex string.
-    fn from_hex(s: &str) -> Result<Self, Self::Error> {
-        Self::from_byte_iter(HexToBytesIter::new(s)?)
+    ///
+    /// Accepts an input string either with `0x` prefix or without, if you require specific handling
+    /// of the prefix see [`Self::from_prefixed_hex`] and [`Self::from_no_prefix_hex`].
+    #[rustfmt::skip]
+    fn from_maybe_prefixed_hex<
+        #[cfg(feature = "alloc")] S: AsRef<str> + Into<String>,
+        #[cfg(not(feature = "alloc"))] S: AsRef<str>,
+    >(s: S) -> Result<Self, Self::Error> {
+        if s.as_ref().starts_with("0x") {
+            Self::from_no_prefix_hex(s.as_ref().trim_start_matches("0x"))
+        } else {
+            Self::from_no_prefix_hex(s)
+        }
+    }
+
+    /// Parses provided string as hex explicitly requiring there to not be a `0x` prefix.
+    ///
+    /// This is not recommended for user-supplied inputs because of possible confusion with decimals.
+    /// It should be only used for existing protocols which always encode values as hex without 0x prefix.
+    #[rustfmt::skip]
+    fn from_no_prefix_hex<
+        #[cfg(feature = "alloc")] S: AsRef<str> + Into<String>,
+        #[cfg(not(feature = "alloc"))] S: AsRef<str>,
+    >(s: S) -> Result<Self, Self::Error> {
+        Self::from_byte_iter(HexToBytesIter::new(s.as_ref())?)
+    }
+
+    /// Parses provided string as hex requiring 0x prefix.
+    ///
+    /// This is intended for user-supplied inputs or already-existing protocols in which 0x prefix is used.
+    #[rustfmt::skip]
+    fn from_prefixed_hex<
+        #[cfg(feature = "alloc")] S: AsRef<str> + Into<String>,
+        #[cfg(not(feature = "alloc"))] S: AsRef<str>,
+    >(s: S) -> Result<Self, PrefixedError<Self::Error>> {
+        use PrefixedError::*;
+
+        if !s.as_ref().starts_with("0x") {
+            #[cfg(feature = "alloc")]
+            return Err(MissingPrefix(MissingPrefixError(s.into())));
+            #[cfg(not(feature = "alloc"))]
+            return Err(MissingPrefix(MissingPrefixError()));
+        } else {
+            Ok(Self::from_no_prefix_hex(s.as_ref().trim_start_matches("0x"))?)
```

Comment:
Same problem as above, you can use `s.as_ref().strip_prefix("0x").ok_or_else(|| MissingPrefixError(s.into()).into())`